firestore-bigquery-export/scripts/gen-schema-view/src/schema.ts

```diff
@@ -100,60 +128,99 @@ export class FirestoreBigQuerySchemaViewFactory {
     let latestView = dataset.table(latestSchemaViewName);
     const [latestViewExists] = await latestView.exists();
 
+    let result = userSchemaView(
+      datasetId,
+      rawChangeLogTableName,
+      firestoreSchema
+    );
+    let bigQueryFields = result.fields;
+    const options = {
+      friendlyName: changeLogSchemaViewName,
+      view: result.viewInfo,
+    };
     if (!viewExists) {
-      const schemaView = userSchemaView(
-        datasetId,
-        rawChangeLogTableName,
-        firestoreSchema
-      );
       logs.bigQuerySchemaViewCreating(
         changeLogSchemaViewName,
         firestoreSchema,
-        schemaView.query
+        result.viewInfo.query
       );
-      const options = {
-        friendlyName: changeLogSchemaViewName,
-        view: schemaView,
-      };
       await view.create(options);
       logs.bigQuerySchemaViewCreated(changeLogSchemaViewName);
     }
 
+    await view.setMetadata({
```

Comment:
setMetadata is only needed once per view, so I think it should be part of `!viewExists` clause.

Same blow.

Reply:
This is intentional, we want to allow user to change their field descriptions (same below). This is the simplest way of accomplishing without introducing more branching here. Also, this allows users to relax columns from "REQUIRED" to "NULLABLE" mode in addition to adding new columns. 

https://cloud.google.com/bigquery/docs/managing-table-schemas

Reply:
Please add a comment to explain those cases.